### PR TITLE
Feature/datastore queries

### PR DESCRIPTION
**Pull request by acron0**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -8,3 +8,37 @@ kafka:
           - zookeeper:zookeeper
         ports:
           - "9092:9092"
+datastore:
+        image: 'mastodonc/kixi.datastore:latest'
```
Comment (elisehuard) [inline docker-compose.yml]: worth checking whether this really updates to latest latest (sometimes you need to force)

File: src/witan/gateway/components/query_router.clj [hunk 2]
```diff
@@ -3,49 +3,48 @@
...
 
-(defn make-graph
-  [service-map]
-  (reduce-kv (fn [a k [fnc fields]]
-               (assoc a (with k (partial fnc service-map)) fields)) {} function-graph))
-
-(defn fix-list-entries
-  [m]
-  (apply hash-map (update-in (first m) [0] #(apply list %))))
+(defn blob->function
```
Comment (elisehuard) [inline src/witan/gateway/components/query_router.clj]: what is blob? :)

File: src/witan/gateway/queries/workspace.clj [hunk 3]
```diff
@@ -32,21 +31,21 @@
   [])
 
 (defn get-workspaces-by-owner
-  [d _ owner]
+  [u d owner]
```
Comment (elisehuard) [inline src/witan/gateway/queries/workspace.clj]: I understand what those are but not a fan of single-letter variables for cognitive effort (I imagine you might change this for final version)

File: docker-compose.yml (hunk 1 above)
Comment (mantree) [inline docker-compose.yml]: you do need to force, know of a way to force it to do so everytime?

File: src/witan/gateway/components/query_router.clj (hunk 2 above)
Comment (mantree) [inline src/witan/gateway/components/query_router.clj]: no idea, i've picked up Ant's half done idea and just tried to get it working.

File: src/witan/gateway/queries/workspace.clj (hunk 3 above)
Comment (mantree) [inline src/witan/gateway/queries/workspace.clj]: yeah, those are horrid. Worse is that they aren't what you think they are!

**Pull request merged**